packages/usa-date-picker/src/index.js

```diff
@@ -1801,8 +1809,18 @@ const handleMouseoverFromDate = (dateEl) => {
   if (hoverDate === currentCalendarDate) return;
 
   const dateToDisplay = parseDateString(hoverDate);
-  const newCalendar = renderCalendar(calendarEl, dateToDisplay);
-  newCalendar.querySelector(CALENDAR_DATE_FOCUSED).focus();
+  const focusedEl = document.activeElement;
+
+  // If the calendar contains a focused element, reapply focus after new calendar renders.
+  // Targeted using aria-label because they are unique and prevents error when
+  // focus is in the table header.
+  if (calendarEl.contains(focusedEl)) {
+    const focusElLabel = focusedEl.getAttribute("aria-label");
+    const newCalendar = renderCalendar(calendarEl, dateToDisplay);
+    newCalendar.querySelector(`[aria-label="${focusElLabel}"]`).focus();
+  } else {
+    renderCalendar(calendarEl, dateToDisplay);
+  }
```

Comment:
From JS comments:

>   If the calendar contains a focused element, reapply focus after new calendar renders.
>  Targeted using aria-label because they are unique and prevents error when focus is in the table header.